OpenRA.Mods.Common/Traits/Interactable.cs

```diff
@@ -52,8 +66,26 @@ Rectangle AutoBounds(Actor self, WorldRenderer wr)
 			return autoBounds.Select(s => s.AutoMouseoverBounds(self, wr)).FirstOrDefault(r => !r.IsEmpty);
 		}
 
+		int2[] PolygonBounds(Actor self, WorldRenderer wr)
+		{
+			var screenVertices = new int2[info.Polygon.Length];
+
+			for (var i = 0; i < info.Polygon.Length; i++)
+			{
+				var vec = info.Polygon[i] + polygonCenterOffset;
+				var offset = new int2(vec.X * wr.TileSize.Width / wr.TileScale, vec.Y * wr.TileSize.Height / wr.TileScale);
+
+				screenVertices[i] = wr.ScreenPxPosition(self.CenterPosition) + offset;
+			}
+
+			return screenVertices;
+		}
+
 		Polygon Bounds(Actor self, WorldRenderer wr, WDist[] bounds)
```

Comment:
Since this is called by `DecorationBounds` this would also mean that `info.Polygon` is used for the decoration bounds as well - was that intended?

Reply:
You're right, it wasn't. I fixed it.